src/Language/PureScript/Names.hs

```diff
@@ -84,6 +88,7 @@ runIdent :: Ident -> Text
 runIdent (Ident i) = i
 runIdent (GenIdent Nothing n) = "$" <> T.pack (show n)
 runIdent (GenIdent (Just name) n) = "$" <> name <> T.pack (show n)
+runIdent UnusedIdent = "__unused"
```

Comment:
Now that this is a special identifier, shouldn't this also be prefixed with a `$` so it can't be mixed up with `Ident "__unused"`

Reply:
good point. Will fix.